### PR TITLE
feat: Discover matching layout data in config/

### PR DESCRIPTION
**Pull request by nickcoutsos**
I guess this is more of a proof of concept pending how configurable you want this to be. The idea is to support automatically detecting locally provided layout data. It follows the convention I'm using in keymap-editor, looking for `<keyboard>.json` instead of `info.json` (I switched to the former a while back to support multiple keymaps in one repo without requiring branching).

Couple of notes:

- It basically ignores `keymap_patterns` and looks exclusively for JSON files under `config/`
- I haven't used keymap-drawer enough to be familiar with how alternate layouts are handled. In keymap-editor I have auto-generated layouts from parsed matrix transforms, and at runtime I look for a chosen node in the keymap. I know that some of the layouts here (corne, for example) follow QMK layout naming conventions but the yaml mapping is only keyed by keyboard name.

### CODE REVIEW

File: .github/workflows/draw-zmk.yml [hunk 1]
```diff
@@ -74,6 +74,11 @@ jobs:
             draw_args=$(get_args "${{ inputs.draw_args }}" "$keyboard")
             echo "INFO:   got extra draw args: $draw_args"
 
+            if [ -f "config/${keyboard}.json" ]; then
+              echo "INFO:   found config/${keyboard}.json";
+              draw_args+=" -j config/${keyboard}.json"
```
Comment (caksoylar) [inline .github/workflows/draw-zmk.yml:79]:
```suggestion
                echo "INFO:   found config/${keyboard}.json";
                draw_args+=" -j config/${keyboard}.json"
```

**Comment by caksoylar** [comment]
Thanks, I think it should help and not hurt in general so I am happy with the current PR. The only potential issue I see is it not being disable-able but I doubt that `$keyboard.json` will be problematic if it exists.

> I haven't used keymap-drawer enough to be familiar with how alternate layouts are handled. In keymap-editor I have auto-generated layouts from parsed matrix transforms, and at runtime I look for a chosen node in the keymap. I know that some of the layouts here (corne, for example) follow QMK layout naming conventions but the yaml mapping is only keyed by keyboard name.

You can see the layout selection logic [here](https://github.com/caksoylar/keymap-drawer/blob/main/keymap_drawer/physical_layout.py#L144): If the json file is not just an array but follows the QMK convention (has `layouts.*`) you can select one of them in the yaml `layout.qmk_layout: LAYOUT_split_3x5_3` or command line `-l/--qmk-layout LAYOUT_split_3x5_3` etc. Otherwise the first defined one will be used.

The parser currently doesn't use the chosen transform in the ZMK keymap, because I don't have a mapping layer from `ZMK transform (five_column_transform)` to `QMK layout (LAYOUT_split_3x5_3)`. So in this case the user would have to provide it via `draw_args` like `corne:'-l LAYOUT_split_3x5_3'`. So don't worry about it, I guess :D

**Comment by nickcoutsos** [comment]
Ok, cool, that's pretty much my approach, and I guess such a case would need user intervention anyway.

**Comment by caksoylar** [comment]
Tested it here and works great, thanks! https://github.com/caksoylar/zmk-config/actions/runs/5360919865/jobs/9726368163

**Pull request merged**